### PR TITLE
Blog App: validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by hamid-murambiwa**
In this project I made the following changes:
- Added the following validations:
- [ ] For the User model:
  - Name must not be blank.
  - PostsCounter must be an integer greater than or equal to zero.
- [ ] For the Post model:
  - Title must not be blank.
  - Title must not exceed 250 characters.
  - CommentsCounter must be an integer greater than or equal to zero.
  - LikesCounter must be an integer greater than or equal to zero.
- Added unit specs for all of your models' methods and validations.
- Added flash messages in the create actions in all your controllers.
- Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -1,15 +1,30 @@
...
+        <%= image_tag(@user.photo, width:'150px', height:'160px', class:'user-img') %>
+        <div class="user-info">
+        <h1><%= @user.name %></h1>
+        <p>Number of posts: <%= @user.posts_counter.to_i %></p>
+        </div>
+    </div>
+    <div class="btn-con">
+        <%= link_to "Create a post", new_user_post_path, class: "post-btn" %>
+    </div>
     <section>
```
Comment (DeliceLydia) [inline app/views/posts/index.html.erb:13]: - [ ] On `posts#create` you should display a flash message if the action is successful or not posts#create that is applied as well on `comments#create`, and `likes#create` actions uses flash messages for a successful and not successful path

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -2,34 +2,34 @@
...
     end
 
     it "renders 'show' template" do
-      expect(response).to render_template('show')
-    end
-
-    it "renders 'show' template with template in body" do
-      expect(response.body).to match('Show')
+      expect(response).to render_template(:show)
     end
```
Comment (DeliceLydia) [inline spec/requests/posts_spec.rb]: - [ ] They are 4 tests that are failing in this file please fix them
test

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @hamid-murambiwa, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Major highlights for your projects are:
- Followed git workflow
- No linter errors
- Validations are working well


## Required Changes ♻️
_Check the comments under the review._



### Keep Rocking!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by dasileker — APPROVED** [review]
### **Hi @hamid-murambiwa,** 


### **Congratulations 👌👏**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**